### DISCUSSION TITLE
MetaMask

### DISCUSSION THREAD

**Discussion by HubeauChou** [WalletConnect/walletconnect-monorepo#3106] [Developer Support]
Connect MetaMask, Reason(code: 5100, message: "Unsupported chains.")

**Comment by HubeauChou** [comment]
[RelayClient]: requestMatchingResponseNotFound - 14:44:24.9120
Reason(code: 5100, message: "Unsupported chains.")
Reason(code: 5100, message: "Unsupported chains.")
Reason(code: 5100, message: "Unsupported chains.")

**Comment by boidushya** [comment]
Hey! Could you elaborate a bit more on which chains you're trying to connect with?

**Reply by HubeauChou** [reply]
let namespaces: [String: ProposalNamespace] = [
            "eip155": ProposalNamespace(
                chains: [
                    Blockchain("eip155:137")!,
                    Blockchain("eip155:1")!
                ],
                methods: [
                    "personal_sign"
                ], events: []
            ),
        ]

**Reply by HubeauChou** [reply]
>>>> response: Response(id: 1689253673340449, topic: "f77b9945775ec9149dc3761cdc57648e9d0e62a0e7a7320e71e0c1bbf9da736c", chainId: Optional("eip155:1"), result: JSONRPC.RPCResult.error(JSONRPC.JSONRPCError(code: -32602, message: "Invalid params", data: nil)))

**Reply by boidushya** [reply]
Where are you getting `Blockchain()` and `ProposalNamespace()` from? Is that a custom implementation?

**Reply by HubeauChou** [reply]
Yes, I just need personal_sign

**Reply by boidushya** [reply]
Can you elaborate on what Blockchain and ProposalNamespace methods do?

**Reply by HubeauChou** [reply]
MetaMask only seems to support this type, I get used to it.

**Reply by HubeauChou** [reply]
An error was reported in the iOS code

**Reply by boidushya** [reply]
Its still a bit unclear right now what your issue is. Can you share the code snippet that you're using to get connected to MetaMask through WalletConnect?
